R/guide-bins.R

```diff
@@ -123,13 +123,25 @@ guide_train.bins <- function(guide, scale, aesthetic = NULL) {
   if (length(breaks) == 0 || all(is.na(breaks))) {
     return()
   }
-  limits <- scale$get_limits()
-  all_breaks <- c(limits[1], breaks, limits[2])
-  bin_at <- all_breaks[-1] - diff(all_breaks) / 2
   # in the key data frame, use either the aesthetic provided as
   # argument to this function or, as a fall back, the first in the vector
   # of possible aesthetics handled by the scale
   aes_column_name <- aesthetic %||% scale$aesthetics[1]
+
+  if (is.numeric(breaks)) {
+    limits <- scale$get_limits()
+    all_breaks <- c(limits[1], breaks, limits[2])
+    bin_at <- all_breaks[-1] - diff(all_breaks) / 2
+  } else {
```

Comment:
The `else` branch is sufficiently complicated that a comments would be helpful. In particular, explain that when `breaks` is not numeric it's expected to be in the `(<lower>, <upper>]` format. I was very confused by the regex until I figured that out. Also, should this format be mentioned somewhere in the documentation?